EasyDotnet.Tool/Services/RoslynService.cs

```diff
@@ -228,4 +228,117 @@ private static MemberDeclarationSyntax CreateTypeDeclaration(Kind kind, string c
+        solution = await workspace.OpenSolutionAsync(projectPath).ConfigureAwait(false);
+      }
+      else if (projectPath.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) ||
+               projectPath.EndsWith(".fsproj", StringComparison.OrdinalIgnoreCase))
+      {
+        var project = await workspace.OpenProjectAsync(projectPath).ConfigureAwait(false);
+        solution = project.Solution;
+      }
+      else
+      {
+        throw new ArgumentException($"Path must be a .sln or project file (.csproj, .fsproj): {projectPath}");
+      }
+    }
+    catch (InvalidOperationException ex)
+    {
+      throw new InvalidOperationException($"Failed to open {(projectPath.EndsWith(".sln") ? "solution" : "project")}: {ex.Message}", ex);
+    }
+    catch (FileNotFoundException ex)
+    {
+      throw new FileNotFoundException($"Project or solution file not found: {ex.Message}", ex);
+    }
+    catch (IOException ex)
+    {
+      throw new IOException($"IO error while opening {(projectPath.EndsWith(".sln") ? "solution" : "project")}: {ex.Message}", ex);
+    }
+    catch (UnauthorizedAccessException ex)
+    {
+      throw new UnauthorizedAccessException($"Access denied to {(projectPath.EndsWith(".sln") ? "solution" : "project")}: {ex.Message}", ex);
+    }
+
+    var allDocuments = solution.Projects.SelectMany(project => project.Documents);
+    var channel = Channel.CreateUnbounded<DiagnosticMessage>();
+    var writer = channel.Writer;
+
+    var parallelTask = Parallel.ForEachAsync(allDocuments,
+      new ParallelOptions { MaxDegreeOfParallelism = Environment.ProcessorCount },
+      async (document, cancellationToken) =>
+      {
+        try
+        {
+          var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+          if (semanticModel == null) return;
+
+          var diagnostics = semanticModel.GetDiagnostics();
+
+          foreach (var diagnostic in diagnostics)
+          {
+            if (!ShouldIncludeDiagnostic(diagnostic, includeWarnings))
+              continue;
+
+            var lineSpan = diagnostic.Location.GetLineSpan();
+            var diagnosticMessage = new DiagnosticMessage(
+              FilePath: diagnostic.Location.SourceTree?.FilePath ?? document.FilePath ?? string.Empty,
+              Range: new(
+                new(lineSpan.StartLinePosition.Line, lineSpan.StartLinePosition.Character),
+                new(lineSpan.EndLinePosition.Line, lineSpan.EndLinePosition.Character)
+              ),
+              Severity: MapSeverity(diagnostic.Severity),
+              Message: diagnostic.GetMessage(),
+              Code: diagnostic.Id,
+              Source: "roslyn",
+              Category: diagnostic.Descriptor.Category
+            );
+
+            await writer.WriteAsync(diagnosticMessage, cancellationToken).ConfigureAwait(false);
+          }
+        }
+        catch (Exception)
+        {
+          // Skip documents that fail to load

```

Comment:
Consider: Inject LogService and add warning log entries for documents that fails to load